examples/ping.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright 2018 Parity Technologies (UK) Ltd.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+//! A basic example demonstrating some core APIs and concepts of libp2p.
+//!
+//! In the first terminal window, run:
+//!
+//! ```sh
+//! cargo run --example ping
+//! ```
+//!
+//! It will print the PeerId and the listening address, e.g. `Listening on
+//! "/ip4/0.0.0.0/tcp/24915"`
+//!
+//! In the second terminal window, start a new instance of the example with:
+//!
+//! ```sh
+//! cargo run --example ping -- /ip4/127.0.0.1/tcp/24915
+//! ```
+//!
+//! The two nodes establish a connection, negotiate the ping protocol
+//! and begin pinging each other.
+
+use futures::{prelude::*, future};
+use libp2p::{ identity, PeerId, ping::{Ping, PingConfig}, Swarm };
+use rand::Rng;
+use std::env;
+use std::time::Duration;
+
+fn main() {
+    env_logger::init();
+
+    // Create a random PeerId.
+    let id_keys = identity::Keypair::generate_ed25519();
+    let peer_id = PeerId::from(id_keys.public());
+    println!("Local peer id: {:?}", peer_id);
+
+    // Create a transport.
+    let transport = libp2p::build_development_transport(id_keys);
+
+    // Create a ping network behaviour with a somewhat randomized ping interval.
+    let int = Duration::from_secs(rand::thread_rng().gen_range(10, 21));
+    let cfg = PingConfig::new().with_interval(int);
+    let behaviour = Ping::new(cfg);
+
+    // Create a Swarm to manage connections established with
+    // the transport and supporting the ping behaviour.
+    let mut swarm = Swarm::new(transport, behaviour, peer_id);
+
+    // Listen on all interfaces and a random, OS-assigned port.
+    let listen_addr = Swarm::listen_on(&mut swarm, "/ip4/0.0.0.0/tcp/0".parse().unwrap()).unwrap();
+    println!("Listening on {:?} (ping interval: {:?})", listen_addr, int);
```

Comment:
Just as a note, this will conflict with #1032

Reply:
Thanks, I'm aware!